content/en/docs/ambient/usage/extend-waypoint-wasm/wasm-module-distribution/index.md

```diff
@@ -0,0 +1,231 @@
+
+{{< text bash >}}
+$ istioctl x waypoint apply --enroll-namespace --wait
+{{< /text >}}
+
+### Verify traffic without WasmPlugin at the waypoint
+
+{{< text bash >}}
+$ kubectl exec deploy/sleep -- curl -s -w "%{http_code}" -o /dev/null http://productpage:9080/productpage
+200
+{{< /text >}}
+
+### Apply WasmPlugin at waypoint proxy
+
+To configure a WebAssembly filter with a remote Wasm module, create a `WasmPlugin` resource targeting the `waypoint` gateway:
+
+{{< text bash >}}
+$ kubectl get gateway
+NAME               CLASS            ADDRESS        PROGRAMMED   AGE
+bookinfo-gateway   istio            172.18.7.110   True         23h
+waypoint           istio-waypoint   10.96.202.82   True         21h
+{{< /text >}}
+
+{{< text bash >}}
+$ kubectl apply -f - <<EOF
+apiVersion: extensions.istio.io/v1alpha1
+kind: WasmPlugin
+metadata:
+  name: basic-auth-at-waypoint
+spec:
+  targetRefs:
+    - kind: Gateway
+      group: gateway.networking.k8s.io
+      name: waypoint # gateway name retrieved from previous step
+  url: oci://ghcr.io/istio-ecosystem/wasm-extensions/basic_auth:1.12.0
+  phase: AUTHN
+  pluginConfig:
+    basic_auth_rules:
+      - prefix: "/productpage"
+        request_methods:
+          - "GET"
+          - "POST"
+        credentials:
+          - "ok:test"
+          - "YWRtaW4zOmFkbWluMw=="
+EOF
+{{< /text >}}
+
+### View the configured WasmPlugin
+
+{{< text bash >}}
+$ kubectl get wasmplugin
+NAME                     AGE
+basic-auth-at-gateway    28m
+basic-auth-at-waypoint   14m
+{{< /text >}}
+
+### Verify the traffic via waypoint proxy
+
+1. Test internal `/productpage` without credentials
+
+    {{< text bash >}}
+    $ kubectl exec deploy/sleep -- curl -s -w "%{http_code}" -o /dev/null http://productpage:9080/productpage
+    401
+    {{< /text >}}
+
+1. Test internal `/productpage` with credentials
+
+    {{< text bash >}}
+    $ kubectl exec deploy/sleep -- curl -s -w "%{http_code}" -o /dev/null -H "Authorization: Basic YWRtaW4zOmFkbWluMw==" http://productpage:9080/productpage
+    200
+    {{< /text >}}
+
+## Apply WasmPlugin for a specific Service using Waypoint
+
+To configure a WebAssembly filter with a remote Wasm module for a specific service, create a `WasmPlugin` resource targeting the `waypoint` that is specifically created for that service:
+
+{{< text bash >}}
+$ istioctl experimental waypoint apply -n default --name reviews-svc-waypoint --wait
+waypoint default/reviews-svc-waypoint applied
+{{< /text >}}
+
+Label the `reviews` service to use the `reviews-svc-waypoint` waypoint:
+
+{{< text bash >}}
+$ kubectl label service reviews istio.io/use-waypoint=reviews-svc-waypoint
+service/reviews labeled
+{{< /text >}}
+
+Any requests from pods in the ambient mode to the `reviews` service will now be routed through the `reviews-svc-waypoint` waypoint.
```

Comment:
Is this correct? I'd expect it would be routed to `reviews` earlier also, before line 156.

Reply:
> Is this correct? I'd expect it would be routed to `reviews` earlier also, before line 156.

This term was copied from https://preliminary.istio.io/latest/docs/ambient/usage/waypoint/#configure-a-service-to-use-a-specific-waypoint . At line 156, this won't be correct without the new config